lib/Paws/Credential/ECSContainerProfile.pm

```diff
@@ -0,0 +1,127 @@
+package Paws::Credential::ECSContainerProfile;
+  use JSON::MaybeXS;
+  use Moose;
+  use DateTime::Format::ISO8601;
+  with 'Paws::Credential';
+
+  my $defaultMetadataUrl = undef;
+
+  my $containerLocalUrl=$ENV{'AWS_CONTAINER_CREDENTIALS_RELATIVE_URI'};
+
+  if ($containerLocalUrl)
+  {
+    $defaultMetadataUrl = "http://169.254.170.2$containerLocalUrl";
```

Comment:
Hi @gadgetjunkie!

Thanks for your contribution! It's very useful to have an ECS credential provider in Paws :)

I've merged your branch on to a local branch in the main repo (https://github.com/pplu/aws-sdk-perl/tree/feature/ecs-credential-provider) so we can get the provider up to a point where we can ship it.

I've done some commits on top, with @Grinzz suggestion applied, some refactoring and some tests.

I'm hoping we can factor out all the commonalities with InstanceProfile so we don't have so much copy and pasted code (although it's not mandatory for release)